procrustes/permutation.py

```diff
@@ -132,92 +132,69 @@ def permutation(
-    # make the permutation matric by setting the corresponding elements to 1
-    p = np.zeros(c.shape)
-    p[(row_ind, col_ind)] = 1
+    # compute permutation matrix using Hungarian algorithm
+    p = _compute_permutation_hungarian(c)
     # compute one-sided permutation error
     error = compute_error(new_a, new_b, p)
 
     return ProcrustesResult(new_a=new_a, new_b=new_b, t=p, error=error)
 
 
 def permutation_2sided(
-        a,
-        b,
-        single=True,
-        pad=False,
-        unpad_col=False,
-        unpad_row=False,
-        translate=False,
-        scale=False,
-        check_finite=True,
-        mode="normal1",
-        iteration=500,
-        tol=1.0e-8,
-        kopt=None,
-        weight=None,
-        lapack_driver="gesvd"
+    a,
+    b,
+    single=True,
+    method="kopt",
+    guess_p1=None,
+    guess_p2=None,
+    pad=False,
+    unpad_col=False,
+    unpad_row=False,
+    translate=False,
+    scale=False,
+    check_finite=True,
+    options=None,
+    weight=None,
+    lapack_driver="gesvd",
 ):
-    r"""Two-sided permutation Procrustes.
+    r"""Perform two-sided permutation Procrustes.
 
     Parameters
     ----------
     a : ndarray
-        The 2d-array :math:`\mathbf{A}_{m \times n}` which is going to be transformed.
+        The 2d-array :math:`\mathbf{A}` which is going to be transformed.
     b : ndarray
-        The 2d-array :math:`\mathbf{B}_{m \times n}` representing the reference.
+        The 2d-array :math:`\mathbf{B}` representing the reference matrix.
     single : bool
-        If true, the two permutation are assumed to be the same (i.e. two-sided permutation
-        Procrustes with one transformation). Further,
-        (1) if the input matrices `a` and `b` are symmetric within the threshold of 1.e-5,
-        undirected graph matching algorithm from [1] will be applied;
-        (2) if the input matrices `a` and `b` are not symmetric, the directed graph
-        matching algorithm from [1] is applied.
-        If false, the two permutation matrices can be different, i.e.
-        it is the two-sided permutation Procrustes with two transformations
-        (known as the regular two-sided permutation Procrustes here). An flip-flop
-        algorithm taken from [2] is used to iteratively solve this problem. Global optima is not
-        guaranteed here and may need to do an additional k-opt search.
-        Default=True.
+        If `True`, the single-transformation Procrustes is performed to obtain :math:`\mathbf{P}`.
+        If `False`, the two-transformations Procrustes is performed to obtain :math:`\mathbf{P}_1`
+        and :math:`\mathbf{P}_2`.
+    method : str, optional
+        The method to solve for permutation matrices. For `single=False`, these include "flip-flop"
+        and "k-opt" methods. For `single=True`, these include "approx-normal1", "approx-normal2",
+        "approx-umeyama", "approx-umeyama-svd", "k-opt", "soft-assign", and "nmf".
```

Comment:
Do we need to add detailed explanations for all the possible arguments, either here or in the manuscript? For example, `nmf` stands for non-negative matrix factorization, but I would assume someone would have to spend some time digging out what is `nmf` and how it works.

Reply:
Yes, the docstring needs updates and extensions (for both arguments & notes sections). I didn't want to do it before you all agree on the refactoring. So, I will do it as the last step.